### PR TITLE
Unit test - onstartup.go

### PR DESCRIPTION
**Pull request by neilnaveen**
### 1. Why is this pull request needed and what does it do?
- Included test for onstartup.go.

Signed-off-by: neil <42328488+neilnaveen@users.noreply.github.com>

### CODE REVIEW

**Comment by neilnaveen** [comment]
@yongtang 👀 please

**Comment by neilnaveen** [comment]
> @yongtang 👀 please

A friendly reminder!

File: core/dnsserver/onstartup_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		}, {
+
+			name: "Test3",
+			args: args{
+				protocol: "udp",
+				addr:     "dns://",
+				zones: map[string]*Config{
+					"example.com": nil},
+			},
+			want: "udpexample.com:dns://",
```
Comment (chrisohaver) [inline core/dnsserver/onstartup_test.go:46]: the expected values are very confusing.

**Comment by chrisohaver** [comment]
@neilnaveen, Thank you for contributing. Why are you adding this?  Did you find an issue with the code?  Are you adding it just to increase reported coverage?

File: core/dnsserver/onstartup_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "Test1",
+			args: args{
+				protocol: "udp",
```
Comment (chrisohaver) [inline core/dnsserver/onstartup_test.go:22]:
```suggestion
				protocol: "proto://",
```

File: core/dnsserver/onstartup_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+			want: "udpexample.com:53",
+		}, {
+			name: "Test2",
+			args: args{
+				protocol: "udp",
+				addr:     "dns://127.0.0.1:4005",
+				zones: map[string]*Config{
+					"example.com": nil},
+			},
+			want: "udpexample.com:4005 on 127.0.0.1",
```
Comment (chrisohaver) [inline core/dnsserver/onstartup_test.go:36]:
```suggestion
			want: "proto://example.com:4005 on 127.0.0.1",
```

File: core/dnsserver/onstartup_test.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		}, {
+
+			name: "Test3",
+			args: args{
+				protocol: "udp",
```
Comment (chrisohaver) [inline core/dnsserver/onstartup_test.go:41]:
```suggestion
				protocol: "proto://",
```

File: core/dnsserver/onstartup_test.go (hunk 1 above)
Comment (chrisohaver) [inline core/dnsserver/onstartup_test.go:46]:
```suggestion
			want: "proto://example.com:dns://",
```

**Review by chrisohaver — CHANGES_REQUESTED** [review]
Ive suggested some changes to make expected values in the test cases to make more sense.

Should also add a test case that tests multiple zones are sorted in the in the output.

**Pull request closed without merging**